src/plugins/intel_gpu/src/plugin/plugin.cpp

```diff
@@ -968,40 +974,101 @@ Parameter Plugin::GetMetric(const std::string& name, const std::map<std::string,
         auto cloned_network = InferenceEngine::details::cloneNetwork(network);
         auto inputs_info = cloned_network.getInputsInfo();
         ICNNNetwork::InputShapes new_shapes;
-        //std::map<std::string, SizeVector>;
-        bool batch_detected = false;
-        for (auto& info : inputs_info) {
-            if (!info.second)
-                continue;
-            InferenceEngine::Layout layout = info.second->getLayout();
-            auto data = info.second->getInputData();
-            if (!data)
-                continue;
-            std::string name = info.second->getInputData()->getName();
-            auto shape = data->getTensorDesc().getDims();
-            if (layout == InferenceEngine::Layout::NCHW ||
-                layout == InferenceEngine::Layout::NHWC ||
-                layout == InferenceEngine::Layout::NCDHW ||
-                layout == InferenceEngine::Layout::NDHWC ||
-                layout == InferenceEngine::Layout::NC)  {
-                shape[0] = base_batch_size;
-                batch_detected = true;
-            } else if (layout == InferenceEngine::Layout::CN) {
-                shape[1] = base_batch_size;
-                batch_detected = true;
-            }
-            new_shapes[name] = shape;
-        }
+
         try {
-            if (batch_detected) { // reshape only for batched layout
-                cloned_network.reshape(new_shapes);
-                GPU_DEBUG_IF(debug_config->verbose >= 1) {
-                    GPU_DEBUG_COUT << "[GPU_MAX_BATCH_SIZE] Reshaped base batch size to " << base_batch_size << std::endl;
+            std::set<std::string> batched_inputs;
+            std::set<std::string> batched_outputs;
+
+            auto function = InferenceEngine::details::cloneNetwork(cloned_network).getFunction();
+            ov::pass::Manager m;
+            m.register_pass<ngraph::pass::InitNodeInfo>();
+            m.register_pass<ov::pass::FindBatch>(false, true);
```

Comment:
should be **true** (to skip the Detection output from shape propagation, otherwise all detection network will be reported as non-batchable) and **false** (to relax dim tracking for the use case when BATCH is set explicitly )

Reply:
what about the AUTO plugin scenario? Dont they need to get maximum batch size regardless of DO, because the DO can be splitted ? Or.. will it be better to have an additional parmerter for this config? track_do

Reply:
Ah, sorry I got those param in opposite way.. thanks for the comment and will update soon!

Reply:
Updated.